### ISSUE TITLE
AreaChart does not apply custom color scale to area fill

### ISSUE THREAD

**Issue by diego-codes** [carbon-design-system/carbon-charts#921]
#### I am submitting a...

- [ ] Feature request
- [ ] Design defect
- [x] Source code defect
- [ ] Demo/documentation defect
- [ ] Other

#### charts version: 
`0.41.14`

### Issue description
The `AreaChart` component does not apply a custom color scale to the area fill.

#### Steps to produce the issue
See CodeSandbox example: https://codesandbox.io/s/carbon-area-chart-color-bug-m9onw?file=/src/index.js:1547-1602

1. Create an `AreaChart` component with a custom color scale in its options object. Ex: 
  ```js
  {
    color: {
      scale: {
        "Dataset 1": "orange",
        "Dataset 2": "magenta"
      }
    }
  }
  ``` 
2. Notice that the custom color scale only applies to the area stroke, not the area fill.

#### Current behavior
The `AreaChart` component only applies a custom color scale to the area stroke, not its fill.

#### Expected behavior
The `AreaChart` component should apply a custom color scale to both the area stroke and fill.

### Screenshot or recording
![image](https://user-images.githubusercontent.com/5973294/103920429-2934b980-50c6-11eb-8708-8f602982fe67.png)

**Comment by theiliad** [comment]
@sophiiae could you take a look please?

**Comment by sophiiae** [comment]
This issue has fixed with #919
